src/ophyd_async/epics/pvi/pvi.py

```diff
@@ -296,7 +301,9 @@ async def fill_pvi_entries(
 
 
 def create_children_from_annotations(
```

Comment:
We definitely need a PVI test that `included_optional_fields` and `device_vectors` args are functioning. They should:
    * check we get the attributes (including device vectors) after `create_children_from_annotations`, but before `connect`
    * check that the device vectors are of the specified size
    * check that after `connect` (mock for test), the signals and devices remain the same by ref that they were after `create_children_from_annotations`